### PR TITLE
Tests: adds unit tests for `pkg/graph/schema/resolver.go`

### PR DESCRIPTION
**Pull request by Rudra-Sankha-Sinhamahapatra**
## Description
Added unit tests for the `NewCombinedResolver` function in the schema package. 
This improves test coverage and validates both success and error paths.

## Changes
- Created `resolver_test.go` with table-driven tests
- Added mock HTTP server to simulate Kubernetes API responses
- Tested successful resolver creation
- Tested error handling when the API server returns errors

## Testing
- All tests pass with `go test ./pkg/graph/schema -v`
- Doesn't require special environment setup like the integration tests

Screenshot 2025-03-17 at 5 02 37 PM

All tested all unit tests everything passes 

  cc: @a-hilaly

### CODE REVIEW

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
@a-hilaly please review this pr

File: pkg/graph/schema/resolver_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/rest"
+)
+
+func TestNewCombinedResolver(t *testing.T) {
```
Comment (a-hilaly) [inline pkg/graph/schema/resolver_test.go]: Can we use a fake client instead?

Comment (a-hilaly) [reply pkg/graph/schema/resolver_test.go]: In fact not sure if it's super useful to test this function

**Review by a-hilaly — COMMENTED** [review]
Thanks @Rudra-Sankha-Sinhamahapatra

File: pkg/graph/schema/resolver_test.go (hunk 1 above)
Comment (Rudra-Sankha-Sinhamahapatra) [reply pkg/graph/schema/resolver_test.go]: @a-hilaly 
Hey I have added fake client instead of http mocking 

And I have tested resolver.go by returning wrong value to ensure resolver_test.go is testing resolver.go correctly 

Yeah currently test cases are passing 

Screenshot 2025-03-19 at 3 08 54 PM


I tried using something wrong in resolver.go then the testcases fails 

So I have tested it and it works fine

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
@a-hilaly  can you check this now

**Review by a-hilaly — COMMENTED** [review]
Hi @Rudra-Sankha-Sinhamahapatra, thanks for this PR!

I don't think these tests add sufficient value as they're just testing thin wrappers around k8s client init with no complex logic or edge cases. any issues with this functionality would be caught by our integration tests that actually use the resolver.

for future packages you'd like to increase test coverage for can you please reach out on our kro channel to discuss it first?

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
Sure thanks
